app/views/goods/_listelement.html.erb

```diff
@@ -1,13 +1,13 @@
 
-<div class="uk-text-center">
+<div class="uk-text-center" width="500px" height="500px" >
         <div class="uk-inline-clip uk-transition-toggle" tabindex="0">
-                      <% if listelement.photo.attached? %>
-                <%= image_tag listelement.thumb_list%>
-              <% else %>
-                <% product = Product.find_by_id(listelement.product_id) %>
-                <% if product.photo.attached? %>
-                  <%= image_tag product.thumb_list%>
-                <% end %> 
+            <% if listelement.photo.attached? %>
```

Comment:
A termékek listázása termékenként 4-szer nyúl adatbázishoz, ami már a seedelt 30 termékkel is kicsit lassúvá teszi a renderelést, sokkal több termékkel pedig még rosszabb lesz a helyzet.

Ezt az okozza, hogy minden egyes termékhez külön kéritek le a photo-t, utána a product-ot, majd végül a product photo-ját.
Ezt a problémát Rails-ben N+1 query-nek hívják, és [mondjuk itt](https://semaphoreci.com/blog/2017/08/09/faster-rails-eliminating-n-plus-one-queries.html) tudtok jobban utána olvasni. A megoldása röviden, hogy már az elején be kell tölteni az összes kapcsolódó rekordot, amire majd később szükségetek lesz egy `includes()` segítségével.

Reply:
Feltöltöttem egy megoldást az n+1 query-re. Active_storage-el nem akart működni az icludes, ezért  Good.with_attached_photo-al csináltam, remélem azzal is jó. A view-ból a képet megkereső logikát átvittem egy helperbe.